src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -213,16 +213,29 @@ public static string ValidateIssuer(string issuer, SecurityToken securityToken,
         /// <returns>The issuer to use when creating the "Claim"(s) in a "ClaimsIdentity".</returns>
         /// <exception cref="ArgumentNullException">If 'validationParameters' is null.</exception>
         /// <exception cref="ArgumentNullException">If 'issuer' is null or whitespace and <see cref="TokenValidationParameters.ValidateIssuer"/> is true.</exception>
-        /// <exception cref="ArgumentNullException">If ' configuration' is null.</exception>
+        /// <exception cref="ArgumentNullException">If 'configuration' is null.</exception>
         /// <exception cref="SecurityTokenInvalidIssuerException">If <see cref="TokenValidationParameters.ValidIssuer"/> is null or whitespace and <see cref="TokenValidationParameters.ValidIssuers"/> is null and <see cref="BaseConfiguration.Issuer"/> is null.</exception>
         /// <exception cref="SecurityTokenInvalidIssuerException">If 'issuer' failed to matched either <see cref="TokenValidationParameters.ValidIssuer"/> or one of <see cref="TokenValidationParameters.ValidIssuers"/> or <see cref="BaseConfiguration.Issuer"/>.</exception>
         /// <remarks>An EXACT match is required.</remarks>
         internal static string ValidateIssuer(string issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
         {
-            ValueTask<string> vt = ValidateIssuerAsync(issuer, securityToken, validationParameters, configuration);
-            return vt.IsCompletedSuccessfully ?
-                vt.Result :
-                vt.AsTask().ConfigureAwait(false).GetAwaiter().GetResult();
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            // Given that this is a synchronous method and to avoid consuming excesive threads
+            // we will favor the synchronous delegates over the asynchronous one.
+            if (validationParameters.IssuerValidator == null
+                && validationParameters.IssuerValidatorUsingConfiguration == null
+                && validationParameters.IssuerValidatorAsync != null)
```

Comment:
This is internal, does this limit the use case in any way?

Reply:
With this change this is no longer internal but public.

Reply:
The property on TokenValidationParameters is still internal, the callers cant assign it to the pipeline.